src/Storages/ExternalStream/Kafka/Kafka.cpp

```diff
@@ -20,6 +27,64 @@ extern const int OK;
 extern const int RESOURCE_NOT_FOUND;
 }
 
+std::vector<std::pair<std::string, std::string>> Kafka::parseProperties(std::string properties)
```

Comment:
Let's refactor out this util and unit test it.

Reply:
Right, my first thought was to use `unordered_map` too. But the only use case for the `properties` is to loop over it, no lookup or update is needed. Thus `vector` is more efficient here. Besides, I think `vector<pair<string, string>>` is also quite readable, it's just the `std::` prefix makes it horrable (probably should add `using` to remove that).

Reply:
using is not a good practice, always include namespace and you will find it is way more maintainable. I think std::unordered_map has benefits like dedup etc.

Reply:
Dedup is not neccessary here, we can just let the latest takes the priority. If we get complains for such behavior, we can fix it then.

Also, this is to be consistent with what's defined in `KafkaLog`:
```cpp
using KConfParams = std::vector<std::pair<String, String>>;
```

And actually, I will just use this `KConfParams` in my code.